enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,710 @@
+extra metadata to those pod definitions, instead of relying on the
+admission hook to do it.
+
+Some of the workloads that will be classified as management are
+outside of the OpenShift release payload. Therefore we will need to
+document this feature so that workload authors can use it. We will
+also identify add-on components that we know our initial customers
+will use and to which we contribute, and make the changes to add the
+labels ourselves, starting with
+
+* performance-addon-operator
+* sriov-network-operator
+* ptp-operator
+* local-storage-operator
+* cluster-logging (fluentd)
+* klusterlet and klusterlet add-ons (RHACM)
+* kubevirt
+* N3000
+
+### User Stories
+
+#### Radio Access Network (RAN) Use Case
+
+In the context of telcocommunications service providers' 5G Radio
+Access Networks, it is increasingly common to see "cloud native"
+implementations of the 5G Distributed Unit (DU) component. Due to
+latency constraints, this DU component needs to be deployed very close
+to the radio antenna for which it is responsible. In practice, this
+can mean running this component on anything from a single server at
+the base of a remote cell tower or in a datacenter-like environment
+serving several base stations.
+
+A hypothetical DU example is an unusually resource-intensive workload,
+requiring 20 dedicated cores, 24 GiB of RAM consumed as huge pages,
+multiple SR-IOV NICs carrying several Gbps of traffic each, and
+specialized accelerator devices. The node hosting this workload must
+run a realtime kernel, be carefully tuned to ensure low-latency
+requirements can be met, and be configured to support features like
+Precision Timing Protocol (PTP).
+
+The most constrained resource for RAN deployments is CPU, so for now
+the focus of this enhancement is managing CPU requirements.
+Kubernetes resource requests for CPU time are measured in fractional
+"[Kubernetes
+CPUs](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu)". For
+bare metal, 1 Kubernetes CPU is equivalent to 1 hyperthread.
+
+Due to the resource-intensive workload, the overhead for platform
+components such as the OpenShift platform and add-on operators is
+severely restricted when compared to other uses. For example, one
+customer has allocated 4 hyperthreads for all components other than
+their own workloads. OpenShift currently requires 7.
+
+### Implementation Details/Notes/Constraints
+
+#### High-level End-to-end Workflow
+
+This section outlines an end-to-end workflow for deploying a cluster
+with management-workload-partitioning enabled and how pods are
+correctly scheduled to run on the management CPU pool.
+
+1. User sits down at their computer.
+2. The user creates their `install-config.yaml`, including extra
+   values in the specifying the CPUs to include in the management CPU
+   set. (See below for details.)
+3. The user runs the installer.
+4. The installer uses the management CPU pool settings to generate an
```

Comment:
Is the expectation that the management CPU pool is specified for both the master machineconfigpool and the worker machineconfigpool? If so, is the same cpuset going to be universally applicable to both machineconfigpools?

Reply:
The next draft includes an updated schema in the next section. The wording in this section is trying to explain how the management CPU pool will be configured as a specific case, so I would like to keep the wording as-is here.